analyses/cell-type-ewings/scripts/singler-workflow/02-run-singler.R

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env Rscript
+
+# This script runs SingleR on a processed SCE object 
+# All tumor cells present in a combined tumor reference is used 
+# as a reference along with BlueprintEncodeData from celldex 
+# Any cells that are from the library being annotated are removed from the reference 
+
+
+project_root <- here::here()
+
+suppressPackageStartupMessages({
+  library(optparse)
+  library(SingleCellExperiment)
+})
+
+option_list <- list(
+  make_option(
+    opt_str = c("--sce_file"),
+    type = "character",
+    help = "Path to processed SCE file to annotate."
+  ), 
+  make_option(
+    opt_str = c("--tumor_reference_file"),
+    type = "character",
+    default = file.path(project_root, "scratch", "tumor-ref-singler.rds"),
+    help = "Path to RDS file with merged object containing all tumor cells. 
+      The `ref_tumor_label` column in the `colData` will be used for the reference labels."
+  ),
+  make_option(
+    opt_str = c("--output_annotations_file"),
+    type = "character",
+    help = "Path to TSV file to save annotations as obtained from `SingleR` for input SCE file."
+  ), 
+  make_option(
+    opt_str = c("-t", "--threads"),
+    type = "integer",
+    default = 4,
+    help = "Number of multiprocessing threads to use."
+  )
+)
+
+# Parse options
+opt <- parse_args(OptionParser(option_list = option_list))
+
+# Set up -----------------------------------------------------------------------
+
+# make sure input files exist 
+stopifnot(
+  "SCE file does not exist" = file.exists(opt$sce_file),
+  "Tumor reference file does not exist" = file.exists(opt$tumor_reference_file)
+)
+
+# create output directory if it doesn't exist
+output_dir <- dirname(opt$output_annotations_file)
+fs::dir_create(output_dir)
+
+# set up multiprocessing params
+if (opt$threads > 1) {
+  bp_param <- BiocParallel::MulticoreParam(opt$threads)
+} else {
+  bp_param <- BiocParallel::SerialParam()
+}
+
+# read in files 
+sce <- readr::read_rds(opt$sce_file)
+tumor_ref <- readr::read_rds(opt$tumor_reference_file)
+
+# Prep references --------------------------------------------------------------
+
+# grab celldex references
+blueprint_ref <- celldex::BlueprintEncodeData(ensembl = TRUE)
+hpca_ref <- celldex::HumanPrimaryCellAtlasData(ensembl = TRUE)
+
+# remove tumor cells from SCE object for annotation 
+library_id <- metadata(sce)$library_id
+sce_tumor_cells <- tumor_ref$barcodes[which(tumor_ref$library_id == library_id)]
+filtered_sce <- sce[ , !(colnames(sce) %in% sce_tumor_cells)]
```

Comment:
I already said this in a separate comment, but I think we may not want to do this filtering; we'd like to be able to compare how these cells are annotated by SingleR as a positive control.